src/main/java/org/pac4j/j2e/filter/RequiresAuthenticationFilter.java

```diff
@@ -81,52 +75,44 @@ public void init(final FilterConfig filterConfig) throws ServletException {
             final Config config = ConfigBuilder.build(configFactoryParam);
             ConfigSingleton.setConfig(config);
         }
-
         this.clientName = getStringParam(filterConfig, Pac4jConstants.CLIENT_NAME, this.clientName);
-        CommonHelper.assertNotNull(Pac4jConstants.CLIENT_NAME, this.clientName);
-
-        this.requireAnyRole = getStringParam(filterConfig, Pac4jConstants.REQUIRE_ANY_ROLE, this.requireAnyRole);
-        this.requireAllRoles = getStringParam(filterConfig, Pac4jConstants.REQUIRE_ALL_ROLES, this.requireAllRoles);
         this.authorizerName = getStringParam(filterConfig, Pac4jConstants.AUTHORIZER_NAME, this.authorizerName);
-
-        this.useSessionForDirectClient = getBooleanParam(filterConfig, Pac4jConstants.USE_SESSION_FOR_DIRECT_CLIENT, this.useSessionForDirectClient);
-        this.allowDynamicClientSelection = getBooleanParam(filterConfig, Pac4jConstants.ALLOW_DYNAMIC_CLIENT_SELECTION, this.allowDynamicClientSelection);
     }
 
     @Override
     protected final void internalFilter(final HttpServletRequest request, final HttpServletResponse response,
                                         final FilterChain chain) throws IOException, ServletException {
 
+        final WebContext context = new J2EContext(request, response);
+
         final Config config = ConfigSingleton.getConfig();
         CommonHelper.assertNotNull("config", config);
         final Clients clients = config.getClients();
         CommonHelper.assertNotNull("clients", clients);
-        this.authorizer = DefaultAuthorizerBuilder.build(this.authorizer, this.authorizerName, config.getAuthorizers(),
-                this.requireAnyRole, this.requireAllRoles);
+        final Client client = clientFinder.find(clients, context, this.clientName);
```

Comment:
Use the default client finder. Can be replaced by another implementation but as it's a seldom (not recommended) use case, it can only be done by overriding the whole class (no setter for this).